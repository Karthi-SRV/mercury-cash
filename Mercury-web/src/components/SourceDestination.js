import React, { Fragment, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { getAirports, analyse } from '../store/actions/AppActions'
import { Select } from 'antd'

const { Option } = Select 

const SourceDestinationComponent = (props) => {

    const { AirportList } = props
    const [AirportListArray, setAirportListArray] = useState([])
    const [ fromStation, setFromStation ] = useState(null)
    const [ toStation, setToStation ] = useState([])
    const [ response, setResponse ] = useState({})

    useEffect(() => {
        fetchAirports()
    }, [])

    useEffect(() => {
        if (AirportList.length) {
            setAirportListArray(AirportList)
        }
    }, [AirportList])

    const fetchAirports = () => {
        props.getAirports()
    }

    const onChangeFromStation = (id) => {
        setFromStation(id)
    }

    const onChangeToStation = (id) => {
        setToStation(id)
    }

    const getAnalyse = () => {
        props.analyse({sourcePort: fromStation, destinationPorts: toStation}, (data) => {
            setResponse(data)
        })
    }

    return (
        <Fragment>
            <div> SHORTEST PATH </div>
            <div>
                <span>FROM:
                    {
                        AirportListArray && <Select placeholder="Select source" onChange={onChangeFromStation}>
                            {
                                AirportListArray.map((data, key) => (
                                    <Option key={key} value={data.id}>{data.name}</Option>
                                ))
                            }
                        </Select>
                    }
                </span>
                <span>TO:
                    {
                        AirportListArray && <Select placeholder="Select destination" mode="multiple" onChange={onChangeToStation}>
                            {
                                AirportListArray.map((data, key) => (
                                    <Option key={key} value={data.id}>{data.name}</Option>
                                ))
                            }
                        </Select>
                    }
                </span>
                <div> Optimal output: {response.optimalOutput}</div>
                <button type='submit' onClick={getAnalyse}>Get</button>
            </div>
        </Fragment>
    )
}

/** @description store data to your component
* @param {object} state redux state object
*/
const mapStateToProps = (state) => {
    return {
        AirportList: state.app.airports,
    }
}

/**
* provide the action creators as props to your component
* @param {object} dispatch redux dispatch object
*/
const mapDispatchToProps = {
    getAirports,
    analyse,
}

// connect our component with store
export default connect(mapStateToProps, mapDispatchToProps)(SourceDestinationComponent)

SourceDestinationComponent.propTypes = {
    getAirports: PropTypes.func,
    analyse: PropTypes.func,
}