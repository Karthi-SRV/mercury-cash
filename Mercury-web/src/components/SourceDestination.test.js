import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import SourceDestination from './SourceDestination'

configure({ adapter: new Adapter() })
const app = shallow(<SourceDestination />)

describe("SourceDestination", () => {
    it("should render my component without any error", () => {
        const app = shallow(<SourceDestination />)
        expect(app).toMatchSnapshot()
    })
})

it('compare the snapshot to render with empty object', () => {
    expect(app).toMatchSnapshot()
})