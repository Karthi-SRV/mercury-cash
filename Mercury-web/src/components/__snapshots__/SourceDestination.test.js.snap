// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceDestination should render my component without any error 1`] = `ShallowWrapper {}`;

exports[`compare the snapshot to render with empty object 1`] = `ShallowWrapper {}`;
